Foundation/NSArray.swift

```diff
@@ -226,7 +226,7 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         for idx in 0..<count {
             let item = _SwiftValue.store(self[idx])
```

Comment:
The cast to `AnyObject?` is needed because this function (`firstObjectCommon`) returns an `Any?`, and there are two forms of `fetch`, one that takes an `AnyObject` and returns `Any`, and one that takes an `AnyObject?` and returns `Any?`. Without the cast, neither version of `fetch` is a better choice than the other because they both result in an implicit coercion to an optional (in one case for the parameter, in the other case for the return value).

This currently compiles without the cast due to a bug in the type checker, but I'm going to be landing a change to the type checker.